lib/IRGen/IRGenMangler.h

```diff
@@ -173,28 +175,66 @@ class IRGenMangler : public Mangle::ASTMangler {
     return finalize();
   }
 
-  std::string mangleOutlinedInitializeWithTakeFunction(const Type t) {
+  std::string
+  mangleOutlinedInitializeWithTakeFunction(const CanType t,
+                                           IRGenModule *mod = nullptr) {
     beginMangling();
-    appendType(t);
-    appendOperator("Wb");
+    auto canType = getCanTypeInContext(t, mod);
+    if (!canType->hasArchetype()) {
+      appendType(canType);
+      appendOperator("Wb", Index(0));
+    } else {
+      auto *generic = canType->getAnyGeneric();
+      assert(generic);
+      appendAnyGenericType(generic);
+      appendOperator("Wb", Index(mod->getCanTypeID(canType)));
```

Comment:
Why not just mangle the type itself, instead of the unique ID? These symbols are not public so they will not affect the binary size

Reply:
You can't mangle the type itself: the new mangler does not support mangling of types that contain archetypes unless you are doing that for debugging. I do not want to go down the debug symbols path.

Reply:
Map the type out of context first to get an interface type, and mangle that